api/client/metricsdebug/client_test.go

```diff
@@ -7,53 +7,51 @@ import (
 	"errors"
 	"time"
 
+	"github.com/golang/mock/gomock"
 	jc "github.com/juju/testing/checkers"
 	gc "gopkg.in/check.v1"
 
-	basetesting "github.com/juju/juju/api/base/testing"
+	basemocks "github.com/juju/juju/api/base/mocks"
 	"github.com/juju/juju/api/client/metricsdebug"
 	apiservererrors "github.com/juju/juju/apiserver/errors"
 	jujutesting "github.com/juju/juju/juju/testing"
 	"github.com/juju/juju/rpc/params"
 	"github.com/juju/juju/state"
-	"github.com/juju/juju/testing"
 	"github.com/juju/juju/testing/factory"
 )
 
 type metricsdebugSuiteMock struct {
-	testing.BaseSuite
-	manager *metricsdebug.Client
+	jujutesting.JujuConnSuite
```

Comment:
I would recommend avoiding jujuconnsuite and mock only the required behaviour instead. (*state.State, *state.StatePool & factory.Factory).